scripts/create-user.js

```diff
@@ -0,0 +1,228 @@
+      return true;
+    }
+  }
+  return 'Please enter valid tags. Between 1 and 5 tags.';
+}
+const questionTags = {
+  type: 'input',
+  name: 'tags',
+  message: 'Please add your tags: (Separate by commas)',
+  validate: validateTags
+};
+
+function validateChannels(answer) {
+  if (answer.length < 1) {
+    return 'You must choose at least one option.';
+  } else if (answer.length > 3) {
+    return 'You must choose maximum three options'
+  }
+  return true;
+}
+const questionChannels = {
+  type: 'checkbox',
+  name: 'channels',
+  message: 'Please add your channels:',
+  choices: [
+    {
+      name: 'Email'
+    },
+    {
+      name: 'Slack'
+    },
+    {
+      name: 'Linkedin'
+    },
+    {
+      name: 'Facebook'
+    },
+    {
+      name: 'Twitter'
+    }
+  ],
+  validate: validateChannels
+};
+
+const questionByChannel = {
+  email: {
+    type: 'input',
+    name: 'Email',
+    message: 'Please add your contact email:',
+    validate: validateEmail,
+    when: (answers) => answers.channels.find((channel) => channel === 'Email')
+  },
+  slack: {
+    type: 'input',
+    name: 'Slack',
+    message: 'Please add your Slack member id:',
+    when: (answers) => answers.channels.find((channel) => channel === 'Slack')
+  },
+  linkedin: {
+    type: 'input',
+    name: 'Linkedin',
+    message: 'Please add your Linkedin user id:',
+    when: (answers) => answers.channels.find((channel) => channel === 'Linkedin')
+  },
+  facebook: {
+    type: 'input',
+    name: 'Facebook',
+    message: 'Please add your Facebook user id:',
+    when: (answers) => answers.channels.find((channel) => channel === 'Facebook')
+  },
+  twitter: {
+    type: 'input',
+    name: 'Twitter',
+    message: 'Please add your Twitter user id:',
+    when: (answers) => answers.channels.find((channel) => channel === 'Twitter')
+  }
+}
+
+const questions = [
+  questionEmail,
+  questionName,
+  questionAvatar,
+  questionTitle,
+  questionDescription,
+  questionCountry,
+  questionTags,
+  questionChannels,
+  questionByChannel.email,
+  questionByChannel.slack,
+  questionByChannel.linkedin,
+  questionByChannel.facebook,
+  questionByChannel.twitter
+];
+
+async function convertAnswersToSchema(answers) {
+  const choices = ['Email', 'Slack', 'Linkedin', 'Facebook', 'Twitter'];
+  for (let answer in answers) {
+    if (answer === 'tags') {
+      answers[answer] = answers[answer].split(',');
+    } else if (choices.includes(answer)) {
+      answers.channels.forEach((element, index) => {
+        if (element === answer) {
+          answers.channels[index] = {
+            type: answer.toLowerCase(),
+            id: answers[answer]
+          }
+        }
+      })
+      delete answers[answer];
+    }
+  }
+  return answers;
+}
+
+async function main() {
+  console.log('Hi! Thanks for adding you or someone else as a mentor!');
+  const answers = await inquirer.prompt(questions);
+  const schema = await convertAnswersToSchema(answers);
```

Comment:
Is `await` needed?

Reply:
Given that `inquirer.prompt` is an async operation, the other option is to use promises,

Reply:
I have no problem with `async / await` I actually a big fan 😀
I'm talking about `convertAnswersToSchema`. As much can see it's a sync function.

Reply:
Yeah, on that function is not required.
That's because I forgot to convert the function to a promise.
I'm a big fan of async/await too.